opm/parser/eclipse/EclipseState/Aquancon.hpp

```diff
@@ -26,52 +26,65 @@
-    namespace{
-        struct AquanconRecord;
-        }
 
     class Aquancon {
         public:
 
-            struct AquanconOutput{
-                int aquiferID;
-                std::vector<size_t> global_index;
-                std::vector<std::shared_ptr<double>> influx_coeff; // Size = size(global_index)
-                std::vector<double> influx_multiplier; // Size = size(global_index)
-                std::vector<int> reservoir_face_dir; // Size = size(global_index)
-                std::vector<int> record_index;
-            };
+        struct AquancCell {
+            int aquiferID;
+            std::size_t global_index;
+            std::pair<bool, double> influx_coeff;
+            double influx_mult;
+            FaceDir::DirEnum face_dir;
 
-            Aquancon(const EclipseGrid& grid, const Deck& deck);
+            AquancCell(int aquiferID_arg, std::size_t gi, const std::pair<bool, double>& ic, double im, FaceDir::DirEnum fd) :
+                aquiferID(aquiferID_arg),
+                global_index(gi),
+                influx_coeff(ic),
+                influx_mult(im),
+                face_dir(fd)
+            {}
 
-            const std::vector<Aquancon::AquanconOutput>& getAquOutput() const;
+            AquancCell() = default;
 
-        private:
+            bool operator==(const AquancCell& other) const {
+                return this->aquiferID == other.aquiferID &&
+                       this->global_index == other.global_index &&
+                       this->influx_coeff == other.influx_coeff &&
+                       this->influx_mult == other.influx_mult &&
+                       this->face_dir == other.face_dir;
+            }
 
-            static std::vector<Aquancon::AquanconOutput> logic_application(const std::vector<Aquancon::AquanconOutput>& original_vector);
+        };
 
-            static void collate_function(std::vector<Aquancon::AquanconOutput>& output_vector,
-                                  std::vector<Opm::AquanconRecord>& m_aqurecord,
-                                  const std::vector<int>& m_aquiferID_per_record, int m_maxAquID);
 
-            static void convert_record_id_to_aquifer_id(std::vector<int>& record_indices_matching_id, int i,
-                                                        const std::vector<int>& m_aquiferID_per_record);
+            Aquancon() = default;
+            Aquancon(const EclipseGrid& grid, const Deck& deck);
+            Aquancon(const std::unordered_map<int, std::vector<Aquancon::AquancCell>>& data);
 
-            // for a cell to be inside reservoir, its indices need to be within the reservoir grid dimension range,
-            // and it needs to be active
-            static bool cellInsideReservoirAndActive(const EclipseGrid& grid, int i, int j, int k);
+            const std::unordered_map<int, std::vector<Aquancon::AquancCell>>& data() const;
+            bool operator==(const Aquancon& other) const;
+            bool active() const;
```

Comment:
No you are right - the `active()` method is not yet in use. But it will be used in the final PR which uses these classes/refactorings to implement aquifer support in the simulator without accessing the deck. Currently the simulator has code like:

```
if (deck.hasKeyword("AQUANCON"))
    ....
```

that will change to:

```
if (aquancon.active()) 
    ....
```